ee/control/control.go

```diff
@@ -17,16 +17,18 @@ type ControlService struct {
 	logger          log.Logger
 	cancel          context.CancelFunc
 	requestInterval time.Duration
-	consumers       map[string]consumer
-	subscribers     map[string][]subscriber
 	fetcher         dataProvider
 	lastFetched     map[string]string
+	consumers       map[string]consumer
+	subscribers     map[string][]subscriber
+	Execute         func() error
+	Interrupt       func(error)
```

Comment:
Why have these as struct variables, and not methods on the struct?

Reply:
Good idea, I've made this change.